src/main/java/seedu/duke/Duke.java

```diff
@@ -0,0 +1,784 @@
+            while (fileScanner.hasNext()) {
+                String line = fileScanner.nextLine();
+                String[] parts = line.split(" \\| ");
+                try {
+                    String type = parts[0];
+                    boolean isDone = parts[1].equals("1");
+                    String description = parts[2];
+                    Task task = null;
+                    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+
+                    switch (type) {
+                        case "T":
+                            task = new Todo(description);
+                            break;
+                        case "D":
+                            if (parts.length < 4) throw new DukeException("Invalid deadline format in file.");
+                            LocalDateTime byDate = LocalDateTime.parse(parts[3], formatter);
+                            task = new Deadline(description, byDate);
+                            break;
+                        case "E":
+                            if (parts.length < 5) throw new DukeException("Invalid event format in file.");
+                            LocalDateTime from = LocalDateTime.parse(parts[3], formatter);
+                            LocalDateTime to = LocalDateTime.parse(parts[4], formatter);
+                            task = new Event(description, from, to);
+                            break;
+                    }
+
+                    if (task != null) {
+                        if (isDone) {
+                            task.markAsDone();
+                        }
+                        tasks.add(task);
+                    }
+                } catch (DukeException | DateTimeParseException e) {
+                    System.out.println("Skipping invalid task: " + line);
+                }
+            }
+        } catch (FileNotFoundException e) {
+            throw new DukeException("File not found: " + e.getMessage());
+        }
+        return tasks;
+
+    }
+    /**
+     * Saves the current tasks to the file.
+     *
+     * @param tasks The TaskList containing tasks to save.
+     */
+    public void save(TaskList tasks) {
+        try (PrintWriter writer = new PrintWriter(new FileWriter(filePath))) {
+            for (int i = 0; i < tasks.getSize(); i++) {
+                Task task = tasks.getTask(i);
+                writer.println(taskToFileString(task));
+            }
+        } catch (IOException e) {
+            System.out.println("An error occurred while saving tasks to file: " + e.getMessage());
+        }
+    }
+    /**
+     * Converts a task to a formatted string for saving to the file.
+     * The format includes the task type, status, description, and any additional information.
+     *
+     * @param task The task to be converted to a string.
+     * @return A formatted string representing the task.
+     */
+    private String taskToFileString(Task task) {
+        String type = task instanceof Todo ? "T" :
```

Comment:
Would it be more readable if "if-else" statements were used instead of the ternary operator?